ledger/narwhal/compact-batch-certificate/src/bytes.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright (C) 2019-2023 Aleo Systems Inc.
+// This file is part of the snarkVM library.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::*;
+
+impl<N: Network> FromBytes for CompactBatchCertificate<N> {
+    /// Reads the compact batch certificate from the buffer.
+    fn read_le<R: Read>(mut reader: R) -> IoResult<Self> {
+        // Read the version.
+        let version = u8::read_le(&mut reader)?;
+        // Ensure the version is valid.
+        if version != 0 {
+            return Err(error("Invalid batch version"));
+        }
+
+        // Read the certificate ID.
+        let certificate_id = Field::read_le(&mut reader)?;
+        // Read the compact batch header.
+        let compact_batch_header = CompactBatchHeader::read_le(&mut reader)?;
+        // Read the number of signatures.
+        let num_signatures = u32::read_le(&mut reader)?;
```

Comment:
we may want to introduce an additional limit here, like in https://github.com/AleoHQ/snarkVM/pull/1985